rs/boundary_node/ic_boundary/src/core.rs

```diff
@@ -345,6 +374,80 @@ pub async fn main(cli: Cli) -> Result<(), Error> {
         vec![Box::new(metrics_runner), Box::new(generic_limiter_runner)];
     runners.append(&mut registry_runners);
 
+    // HTTP Logs Anonymization
+    let tracker = (cli.misc.crypto_config)
+        .zip(cli.obs.obs_log_anonymization_canister_id)
+        .map(|(crypto_config, log_anonymization_cid)| {
+            let c = Arc::new(CryptoComponent::new(
```

Comment:
@fspreiss As far as I'm aware, we just decorate our main with `#[tokio::main]`, we don't do any custom tokio runtime setup for `ic-boundary`. Also, I'm confused because I didn't run into any of these associated panics while running this locally. Any ideas if I'm just missing something?

Reply:
Ah, yes, this `#[tokio::main]` was what I was looking for. `#[tokio::main]` is just [syntactic sugar](https://docs.rs/tokio/latest/tokio/attr.main.html#using-the-multi-thread-runtime) for creating a multi-threaded runtime and calling `block_on` on it, so we definitely have nested `block_on`s here (the other one being [this one](https://github.com/dfinity/ic/blob/master/rs/crypto/internal/crypto_service_provider/src/vault/remote_csp_vault/tarpc_csp_vault_client.rs#L199) in the CryptoComponent). I'm surprised you didn't see runtime panics ("Cannot start a runtime from within a runtime"): are you sure you called `CryptoComponent::new` with a crypto config that has a unix_socket as vault type?

Reply:
Ah - I did not - in my local testing I was using `InReplica`. Ok, I will make the necessary change, thanks for the warning!

Reply:
Ok, change made - please let me know if it looks OK to you. I'm using `block_in_place` but it's just as part of this initialization code, it is not used again after that. The reason I didn't use `spawn_blocking` is because the context in which this is called is not async (it's in a map callback, and although I could probably find some async map, I figured maybe this is already fine?)

Reply:
@fspreiss per your request, the code was refactored a bit so that we use `spawn_blocking` instead of `block_in_place`. Similarly to the orchestrator, we still use `block_in_place` when calling `sign_basic`.